http-core/src/main/scala/org/apache/pekko/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -307,17 +307,20 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
         s"HTTP chunk extension length exceeds configured limit of ${settings.maxChunkExtLength} characters")
 
     @tailrec def parseSize(cursor: Int, size: Long): StateResult =
-      if (size <= settings.maxChunkSize) {
+      if (size <= Int.MaxValue) {
         byteChar(input, cursor) match {
           case c if CharacterClasses.HEXDIG(c) => parseSize(cursor + 1, size * 16 + CharUtils.hexValue(c))
-          case ';' if cursor > offset          => parseChunkExtensions(size.toInt, cursor + 1)()
+          case c if size > settings.maxChunkSize =>
+            failEntityStream(
+              s"HTTP chunk of $size bytes exceeds the configured limit of ${settings.maxChunkSize} bytes")
+          case ';' if cursor > offset => parseChunkExtensions(size.toInt, cursor + 1)()
           case '\r' if cursor > offset && byteChar(input, cursor + 1) == '\n' =>
             parseChunkBody(size.toInt, "", cursor + 2)
           case '\n' if cursor > offset      => parseChunkBody(size.toInt, "", cursor + 1)
           case c if CharacterClasses.WSP(c) => parseSize(cursor + 1, size) // illegal according to the spec but can happen, see issue #1812
           case c                            => failEntityStream(s"Illegal character '${escape(c)}' in chunk start")
         }
-      } else failEntityStream(s"HTTP chunk size exceeds the configured limit of ${settings.maxChunkSize} bytes")
+      } else failEntityStream(s"HTTP chunk size exceeds Integer.MAX_VALUE (${Int.MaxValue}) bytes")
```

Comment:
I don't think we can have such a big chunk in memory, and if we do want to know how many bytes still there in the chunk, we can just logging:` there is still $n bytes remaining maybe maybe more, increase `settings.maxChunkSize` if you want to parse the full chunk`.

Parse it until the `Int.MaxValue` seems some kind of ddos for me, @jrudolph cc

Reply:
Note that this does not start parsing the actual chunk body. It only reads the size value that's written in the chunk header and bails if that's larger than `Int.MaxValue`. It will never attempt to parse anything larger than `settings.maxChunkSize`.

Reply:
And eg, In Netty, there is a `maxContentLength` to be configured and user can config how to response the when oversized with `handleOversizedMessage`.see : `HttpObjectAggregator`